internal/tiltfile/tiltfile_test.go

```diff
@@ -1127,13 +1170,37 @@ func (f *fixture) yaml(path string, entities ...k8sOpts) {
 			if e.image != "" {
 				s = strings.Replace(s, testyaml.SnackImage, e.image, -1)
 			}
+			if len(e.templateLabels) > 0 {
+				var newLabels string
+				for k, v := range e.templateLabels {
+					newLabels = fmt.Sprintf("%s        %s: %v\n", newLabels, k, v)
+				}
+				s = strings.Replace(s, testyaml.SnackTemplateLabel, newLabels, -1)
+			}
 			s = strings.Replace(s, testyaml.SnackName, e.name, -1)
 			objs, err := k8s.ParseYAMLFromString(s)
 			if err != nil {
 				f.t.Fatal(err)
 			}
 
 			entityObjs = append(entityObjs, objs...)
+		case serviceHelper:
+			s := testyaml.DoggosServiceYaml
```

Comment:
i'm surprised you didn't use the InjectLabels function, which seems a bit more robust

Reply:
Oh whoops I forgot about that -- tho it might be more of a pain to take out the existing labels with that?

Reply:
ah, i'm a goof. this was fine, used the yaml manipulation funcs to do this in my latest commit